Setup/Patch/Data/MigrateSuggestionsConfigPatch.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Setup\Patch\Data;
+
+use Algolia\AlgoliaSearch\Helper\Configuration\AutocompleteHelper;
+use Algolia\AlgoliaSearch\Model\Source\Suggestions;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Framework\Setup\Patch\PatchInterface;
+
+class MigrateSuggestionsConfigPatch implements DataPatchInterface
+{
+    public function __construct(
+        protected ModuleDataSetupInterface $moduleDataSetup,
+    ) {}
+
+    /**
+     * @inheritDoc
+     */
+    public function apply(): PatchInterface
+    {
+        $this->moduleDataSetup->getConnection()->startSetup();
+
+        $this->moveIndexingSettings();
+
+        $this->moduleDataSetup->getConnection()->endSetup();
+
+        return $this;
+    }
+
+    /**
+     * Migrate legacy configurations
+     * @return void
+     */
+    protected function moveIndexingSettings(): void
+    {
+        $connection = $this->moduleDataSetup->getConnection();
+        $configDataTable = $this->moduleDataSetup->getTable('core_config_data');
+
+        // Get current number of configured Magento suggestions
+        $whereConfigPathFrom = $connection->quoteInto('path = ?', AutocompleteHelper::NB_OF_QUERIES_SUGGESTIONS);
+        $select = $connection->select()
+            ->from($configDataTable)
+            ->where($whereConfigPathFrom);
```

Comment:
```suggestion
            ->where('path = ?', AutocompleteHelper::NB_OF_QUERIES_SUGGESTIONS);
```

I realize there's no danger here but I'm wondering if we could we bypass the static analysis warning by avoiding `quoteInto`?